server/app/auth/CiviFormProfileData.java

```diff
@@ -26,6 +27,26 @@ public CiviFormProfileData(Long accountId) {
     this.setId(accountId.toString());
   }
 
+  /**
+   * Sets the "canonical" email field in the profile data. Some identity providers use non-standard
+   * attribute names for email. We use the attribute name provided by pac4j here to ensure all
+   * profiles store the email in the same place to make it is accessible via {@code
+   * CommonProfile.getEmail()}.
+   */
+  public CiviFormProfileData setEmail(String email) {
```

Comment:
**praise:** This is a clever approach: using a well-known location in the `pac4j` profile.

Reply:
thanks!